pkg/file/file_test.go

```diff
@@ -47,6 +77,64 @@ func TestLocalFileOpen(t *testing.T) {
 	}
 }
 
+func TestOpen_Combined(t *testing.T) {
+	openTestCases := []struct {
+		desc          string
+		fileName      string
+		fileMode      int
+		mockFetchFunc func(*os.File) error
+		expectedError bool
+		expectedMode  int
+	}{
+		{
+			desc:          "Success: Download and open in READ mode",
+			fileName:      "testfile.txt",
+			fileMode:      fetchLocalFileMode(READ),
+			mockFetchFunc: func(fd *os.File) error { return nil },
+			expectedError: false,
+			expectedMode:  fetchLocalFileMode(READWRITE),
+		},
+		{
+			desc:          "Error: Download fails",
+			fileName:      "another.doc",
+			fileMode:      fetchLocalFileMode(READ),
+			mockFetchFunc: func(fd *os.File) error { return errors.New("mocked error") },
+			expectedError: true,
+			expectedMode:  fetchLocalFileMode(READWRITE),
+		},
+		{
+			desc:          "Error: Unexpected EOF during download but fileMode is Append",
+			fileName:      "bigfile.zip",
+			fileMode:      fetchLocalFileMode(APPEND),
+			mockFetchFunc: func(fd *os.File) error { return io.EOF },
+			expectedError: false,
+			expectedMode:  1090,
+		},
+	}
+
+	for _, tc := range openTestCases {
```

Comment:
Also, we are not using the iterator, it can help us know which test case failed. Please take a look, if that should be included.

Reply:
Sure will include iterator too when logging errors